lib/ansible/modules/database/postgresql/postgresql_set.py

```diff
@@ -0,0 +1,518 @@
+  returned: always
+  type: dict
+  sample: { "value": 67108864, "unit": "b" }
+context:
+  description:
+  - PostgreSQL setting context.
+  returned: always
+  type: str
+  sample: user
+'''
+
+PG_REQ_VER = 9.4
+
+from copy import deepcopy
+
+try:
+    import psycopg2
+    HAS_PSYCOPG2 = True
+except ImportError:
+    HAS_PSYCOPG2 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.database import SQLParseError
+from ansible.module_utils.postgres import postgres_common_argument_spec
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_native
+
+
+# To allow to set value like 1mb instead of 1MB, etc:
+POSSIBLE_SIZE_UNITS = ("mb", "gb", "tb")
+
+# ===========================================
+# PostgreSQL module specific support methods.
+#
+
+
+def param_get(cursor, module, name):
+    query = ("SELECT name, setting, unit, context, boot_val "
+             "FROM pg_settings WHERE name = '%s'" % name)
+    try:
+        cursor.execute(query)
+        info = cursor.fetchall()
+        cursor.execute("SHOW %s" % name)
+        val = cursor.fetchone()
+
+    except Exception as e:
+        module.fail_json(msg="Unable to get %s value due to : %s" % (name, to_native(e)))
+
+    raw_val = info[0][1]
+    unit = info[0][2]
+    context = info[0][3]
+    boot_val = info[0][4]
+
+    if val[0] == 'True':
+        val[0] = 'on'
+    elif val[0] == 'False':
+        val[0] = 'off'
+
+    if unit == 'kB':
+        raw_val = int(raw_val) * 1024
+        boot_val = int(boot_val) * 1024
+        unit = 'b'
+
+    elif unit == 'MB':
+        raw_val = int(raw_val) * 1024 * 1024
+        boot_val = int(boot_val) * 1024 * 1024
+        unit = 'b'
+
+    return (val[0], raw_val, unit, boot_val, context)
+
+
+def pretty_to_bytes(pretty_val):
+    # The function returns a value in bytes
+    # if the value contains 'B', 'kB', 'MB', 'GB', 'TB'.
+    # Otherwise it returns the passed argument.
+
+    val_in_bytes = None
+
+    if 'kB' in pretty_val:
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024
+
+    elif 'MB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024
+
+    elif 'GB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024 * 1024
+
+    elif 'TB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024 * 1024 * 1024
+
+    elif 'B' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part
+
+    else:
+        return pretty_val
+
+    return val_in_bytes
+
+
+def param_set(cursor, module, name, value):
+    try:
+        if value in ('DEFAULT', 'default'):
```

Comment:
How about changing this condition to `if lower(value) == 'default'` ?

Reply:
+